application/apps/indexer/session/src/handlers/search.rs

```diff
@@ -34,6 +34,22 @@ pub async fn execute_search(
     state: SessionStateAPI,
 ) -> OperationResult<u64> {
     debug!("RUST: Search operation is requested");
+    let invalid = filters
+        .iter()
+        .filter_map(|f| if f.valid() { None } else { Some(&f.value) })
+        .cloned()
+        .collect::<Vec<String>>();
+    if !invalid.is_empty() {
+        Err(NativeError {
+            severity: Severity::ERROR,
+            kind: NativeErrorKind::OperationSearch,
+            message: Some(format!(
+                "Next {} filter(s) are invalid: {}",
+                invalid.len(),
+                invalid.join("; ")
+            )),
+        })?;
+    }
```

Comment:
why not move this check to `searchers.rs` and execute it when calling `holder.setup`?

Reply:
Motivation is: holder isn't shared "above" threads, it stored in `SessionState` and before use it, it should be taken from state and after usage I have to put it back. To avoid these steps (1. take holder, 2. try setup. 3. put it back if error) I've putted check here. Also we are checking a little different things: filter and search condition for values-extraction. In the first case we are considering flags, in the second - only regex as a string.

Well if we put this check inside a holder we will need more steps todo then now.